jenkins/common.groovy

```diff
@@ -0,0 +1,180 @@
+
+
+VETSGOV_BUILDTYPES = [
+  'development',
+  'staging',
+  'production'
+]
+
+VAGOV_BUILDTYPES = [
+  'vagovdev',
+  'vagovstaging',
+  'vagovprod'
+]
+
+DRUPAL_MAPPING = [
+  'dev': 'vagovdev',
+  'staging': 'vagovstaging',
+  'live': 'vagovprod',
+]
+
+DEV_BRANCH = 'master'
+STAGING_BRANCH = 'master'
+PROD_BRANCH = 'master'
+
+IS_DEV_BRANCH = env.BRANCH_NAME == DEV_BRANCH
+IS_STAGING_BRANCH = env.BRANCH_NAME == STAGING_BRANCH
+IS_PROD_BRANCH = env.BRANCH_NAME == PROD_BRANCH
+
+def isReviewable() {
+  return !IS_DEV_BRANCH && !IS_STAGING_BRANCH && !IS_PROD_BRANCH
+}
+
+def isDeployable() {
+  return (IS_DEV_BRANCH ||
+          IS_STAGING_BRANCH) &&
+    !env.CHANGE_TARGET &&
+    !currentBuild.nextBuild // if there's a later build on this job (branch), don't deploy
+}
+
+def shouldBail() {
```

Comment:
It seems like there might be a more descriptive name for this function, since we bail out of certain stages for multiple reasons. `hasLaterBuild` or something?

Reply:
Yeah and it seems like `isReviewable()` has similar logic that could be reused/renamed. I'll do another pass and try to make things more decipherable.

Reply:
I've been looking over the logic in these few functions that contain logic about whether to bail or not and I'm not sure if they make sense to me anymore or I've just been looking at this for too long. Why is `!IS_DEV_BRANCH && !IS_STAGING_BRANCH && !IS_PROD_BRANCH` included in the shouldbail function?

Reply:
`shouldBail` is a weird one, and should probably be separated out into separate conditions or at least a more expressive function. 

My understanding is that if we're on a dev/staging/prod branch (which in our case is always master but sometimes there are special circumstances where we build dev off of a feature branch instead), we always want our build to complete and be archived into S3. 

However, if we're on a different branch (anything other than master) then we're likely in a feature branch/pull request and pushing new updates regularly. In this case, if a new commit is pushed before the former is finished building, we cancel the old job. I believe we do this because we aren't concerned about archiving those intermediary commits on feature branches, and it conserves resources on Jenkins. This condition is detected by ` currentBuild.nextBuild` - if that's true, then there's a more recent commit on the Jenkins queue.

Reply:
That was to protect us from aborting the build on the master branch. There's an option to set a branch for each environment (defaults to master for all), so we check each of them.